### ISSUE TITLE
HTTPClient.query_string_from_dict does not handle UTF8.

### ISSUE THREAD

**Issue by vicguedez** [godotengine/godot#17012]
**Godot version:**
3.1-custom_build @ Debian 9

**Issue description:**
This code:
```
var client = HTTPClient.new()

var query = client.query_string_from_dict({"test": "¥ · £ · € · $ · ¢ · ₡ · ₢ · ₣ · ₤ · ₥ · ₦ · ₧ · ₨ · ₩ · ₪ · ₫ · ₭ · ₮ · ₯ · ₹"})

print(query)
```

Produces:

```
test=%FF%FF%20%FF%FF%20%FF%FF%20%FF%FF%20%FF%FF%FF%20%FF%FF%20%24%20%FF%FF%20%FF%FF%20%FF%FF%20%FF%FF%FF%20%FF%FF%20%FF%FF%FF%20%FF%FF%20%FF%FF%FF%20%FF%FF%20%FF%FF%FF%20%FF%FF%20%FF%FF%FF%20%FF%FF%20%FF%FF%FF%20%FF%FF%20%FF%FF%FF%20%FF
```
Which is not encoded nor usable.

**Comment by VitaZheltyakov** [comment]
https://github.com/godotengine/godot/issues/17875
Already decided

**Comment by KoBeWi** [comment]
Closing as fixed in #17583
